### PR TITLE
chore: Add error in EchoArray for regular objects 

### PR DESCRIPTION
**Pull request by mykola-vrmchk**

### CODE REVIEW

**Comment by mykola-vrmchk** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2631** <a href="https://app.graphite.dev/github/pr/dxos/dxos/2631" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dxos/dxos/2631?utm_source=stack-comment).

File: packages/core/echo/echo-schema/src/echo-array.ts [hunk 1]
```diff
@@ -330,6 +330,14 @@ export class EchoArray<T> implements Array<T> {
     ) {
       return new Reference((value as any)['@id']);
     } else {
+      assert(
+        value === null ||
+          value === undefined ||
+          typeof value === 'boolean' ||
+          typeof value === 'number' ||
+          typeof value === 'string',
+        'Not EchoObject or primitive value'
```
Comment (richburdon) [inline packages/core/echo/echo-schema/src/echo-array.ts:339]: `Invalid type: ${JSON.stringify(value)}`

Short errors with data.

File: packages/core/echo/echo-typegen/src/codegen.ts [hunk 2]
```diff
@@ -125,20 +125,10 @@ export const generate = (root: pb.NamespaceBase): string => {
...
-    yield text`
-      export namespace ${ns.name} {
-        ${ns.nestedArray.flatMap((nested) => Array.from(emitDeclarations(nested)))}
-      }
-      
-    `;
-  }
-
   if (ns instanceof pb.Type) {
     if (ns.name === 'Text') {
```
Comment (richburdon) [inline packages/core/echo/echo-typegen/src/codegen.ts:133]: extract any string literals that are used more than once?

**Review by richburdon — APPROVED** [review]

File: packages/core/echo/echo-schema/src/echo-array.ts [hunk 3]
```diff
@@ -330,6 +330,14 @@ export class EchoArray<T> implements Array<T> {
     ) {
       return new Reference((value as any)['@id']);
     } else {
+      assert(
```
[the remark this answers is not in the document]
Comment (mykola-vrmchk) [reply packages/core/echo/echo-schema/src/echo-array.ts:333]: > above `=== object should mean !== null`?
There is a check for that, or I am misunderstanding the comment

[the remark this answers is not in the document]
Comment (mykola-vrmchk) [reply packages/core/echo/echo-schema/src/echo-array.ts:333]: > also why do we test the length of the number of properties?

I don't know for sure, I guess, it is some kind of Proxy detection (@dmaretskyi ?)

File: packages/core/echo/echo-typegen/src/codegen.ts (hunk 2 above)
Comment (mykola-vrmchk) [reply packages/core/echo/echo-typegen/src/codegen.ts:133]: Sure

**Pull request merged**